MANIFEST.in

```diff
@@ -0,0 +1,20 @@
+
```

Comment:
> Who builds distribution tarballs anyway? Developers do. And developers have checkouts.

Automated build tools/scripts (and users) might. In particular scripts/tools downloading/building through prominently displayed tarballs available from the github project's main and releases page (which are not sdist generated).

This might not that big of a deal if the information provided by MANIFEST.in/setuptools-git  would only affect run of sdist, but it also affects (infects) how `package_data` and `include_package_data` is interpreted by setuptools, and that has consequences on what gets installed and included in wheels.

> Consider that if git was more prominently around when the whole Python packaging debacle became known as setuptools, we probably wouldn't be using MANIFEST.in in the first place!

There were plenty of revision control systems in existence (an prominent use) back then. How would git change anything?

An sdist distribution can exclude most of the files under revision control, can include generated files ...
If sdist didn't have MANIFEST.in like functionality it would be necessary to invent it.

As it stands with setuotools-git the generated .tar.gz weigh-in at aprox. 35MB. With a suitably written MANIFEST.in it shrinks to ~3MB. I know at least a while ago a limit for a PyPI distributed file was ~40MB.

Also note that currently .pyx .c .cpp files are installed/included in .whl files (courtesy of setuptools-git), as well as (possibly multiple python abi tagged) .{so,pyd} files that are result of build_ext -i (but that one is find_package_data's (use) fault).